plugins/in_tail/tail_fs_inotify.c

```diff
@@ -289,6 +289,33 @@ static int tail_fs_event(struct flb_input_instance *ins,
     return 0;
 }
 
+static int in_tail_progress_check_callback(struct flb_input_instance *ins,
+                                           struct flb_config *config, void *context)
+{
+    int ret = 0;
+    struct mk_list *tmp;
+    struct mk_list *head;
+    struct flb_tail_config *ctx = context;
+    struct flb_tail_file *file;
+    struct stat st;
+    (void) config;
+
+    mk_list_foreach_safe(head, tmp, &ctx->files_event) {
+        file = mk_list_entry(head, struct flb_tail_file, _head);
+
```

Comment:
what about to try to avoid the syscall ? (if possible)

```c
if (file->offset < file->size) {
    flb_tail_file_chunk(file);
    continue;
}
```

Reply:
Which syscall?

Reply:
I'll run a test to see if the call to fstat that happens after this code is unnecessary

Reply:
Looking at it again it makes more sense. The heat has been a little killer down here lately.

Reply:
I also added code to reset the `file->size` when fstat is called during the progress check interval.